plugins/scanners/scanoss/src/main/kotlin/ScanOssResultParser.kt

```diff
@@ -23,8 +23,11 @@ import com.scanoss.dto.ScanFileDetails
 import com.scanoss.dto.ScanFileResult
 import com.scanoss.dto.enums.MatchType
 
+import java.lang.invoke.MethodHandles
```

Comment:
Commit message: Is this really a `fix`, or should it be a `feat`?

Reply:
You raise a good point. I agree that 'fix' isn't the most accurate descriptor. 

This is more of a structural improvement to how we handle match data. I propose changing the commit message to:

```
refactor(scanoss): Improve snippet generation with primary PURL approach

Replace cartesian product approach (one snippet per PURL-location pair)
with one snippet per location using primary PURL as identifier and storing
all related PURLs in metadata.

``` 

I'll proceed if you agree

Reply:
You raise a good point. I agree that 'fix' isn't the most accurate descriptor.

This is more of a structural improvement to how we handle match data. I propose changing the commit message to:

```
refactor(scanoss): Improve snippet generation with primary PURL approach

Replace cartesian product approach (one snippet per PURL-location pair)
with one snippet per location using primary PURL as identifier and storing
all related PURLs in metadata.
```

I'll proceed if you agree